lib/src/arg_results.dart

```diff
@@ -73,6 +73,51 @@ class ArgResults {
     return option.valueOrDefault(_parsed[name]);
   }
 
+  /// Returns the parsed or default command-line flag named [name].
+  ///
+  /// [name] must be a valid flag name in the parser.
+  bool flag(String name) {
+    if (!_parser.options.containsKey(name)) {
+      throw ArgumentError('Could not find an option named "$name".');
+    }
+
+    var option = _parser.options[name]!;
```

Comment:
I would assign this above, then check for null instead of the `containsKey` call. This will do one less map lookup and give you type promotion instead of having to rely on `!`.